Clarifying the behavior of PSA sync

bergerhoffer opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS-<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch. --->

Version(s):
4.11+

Issue:
N/A

Link to docs preview:
http://file.rdu.redhat.com/~ahoffer/2022/psa-clarifications/authentication/understanding-and-managing-pod-security-admission.html#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

<!-- NOTE:
Automatic preview functionality is currently not available.
  * OpenShift documentation team members (core and aligned) must include a link to a locally generated preview for PRs that update the rendered build in any way.
  * External contributors can request a generated preview from the OpenShift documentation team. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->



<!--- Next steps after opening your PR:

* Ask for peer review from the OpenShift docs team:
  - For Red Hat associates: Ping @peer-review-squad requesting a review in the #forum-docs-review channel (CoreOS Slack workspace) and provide the following information:
    * A link to the PR.
    * The size of the PR that the GitHub bot assigns (ex: XS, S, M, L, XL).
    * If there is urgency or a deadline for the review.
  - For community authors: Request a review by tagging @openshift/team-documentation in a GitHub comment.

  Slack is the quickest and preferred way to request a review.

* IMPORTANT:
  - All documentation changes must be verified by a QE team associate before merging.
  - Squash to one commit before submitting your PR for peer review.

* For more information about verifying your content, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#verification-of-your-content

* For more information about contributing to OpenShift documentation, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/contributing.adoc

Additional resources

The OpenShift docs repo adheres to the following style guides:

- OpenShift documentation guidelines (OSDOCS)
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc
- Red Hat Supplementary Style Guide (SSG)
  https://redhat-documentation.github.io/supplementary-style-guide/
- Modular Documentation Reference Guide (Mod Docs)
  https://redhat-documentation.github.io/modular-docs/
- IBM Style Guide (ISG)
  https://www.ibm.com/docs/en/ibm-style

You can log in to the ISG by using your @redhat.com id and single sign-on (SSO) credentials. --->

bergerhoffer commented:
@stlaz Can you please look?

http://file.rdu.redhat.com/~ahoffer/2022/psa-clarifications/authentication/understanding-and-managing-pod-security-admission.html#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

bscott-rh commented on modules/security-context-constraints-psa-synchronization.adoc:
```diff
@@ -18,18 +18,14 @@ Applying pods directly might use the SCC privileges of the user who runs the pod
 * `openshift`
+* All system-created namespaces that are prefixed with `openshift-`, except for `openshift-operators`
+
+By default, all namespaces that have an `openshift-` prefix are not synchronized. You can enable synchronization for any user-created [x-]`openshift-*` namespaces by setting the value of the `security.openshift.io/scc.podSecurityLabelSync` label to `true`. You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for `openshift-operators`.
```
```suggestion
By default, {product-title} will not synchronize namespaces that have an `openshift-` prefix. 
```
Does it make sense to reword this to avoid passive voice?

stlaz replied:
The original passive voice still sounds the best to me here

stlaz commented:
LGTM

bergerhoffer commented:
@xingxingxia Can you please QE review this update?

http://file.rdu.redhat.com/~ahoffer/2022/psa-clarifications/authentication/understanding-and-managing-pod-security-admission.html#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

xingxingxia commented:
/lgtm
However, for `"Security context constraint synchronization with pod security standards"` section and `"Controlling pod security admission synchronization"` section, one has an `"Important"` note, the other has a `"NOTE"` note. Though with a bit not exactly same wording, they actually intend to tell the totally same explanation about the different namespaces. I understand they are in different `.adoc` modules, but from user end, the explanation shows twice in the same page `"Understanding and managing pod security admission"`, won't they look duplicate from within one same page?

bergerhoffer commented:
> /lgtm However, for `"Security context constraint synchronization with pod security standards"` section and `"Controlling pod security admission synchronization"` section, one has an `"Important"` note, the other has a `"NOTE"` note. Though with a bit not exactly same wording, they actually intend to tell the totally same explanation about the different namespaces. I understand they are in different `.adoc` modules, but from user end, the explanation shows twice in the same page `"Understanding and managing pod security admission"`, won't they look duplicate from within one same page?

@xingxingxia Good catch! Not sure how I didn't notice that similar note right below. We do sometimes want to have the important notes in each section, because they are modules and technically could be viewed separately or on another page sometime. But I agree that back-to-back like this looks very duplicative.

I've moved this detailed note down into the actual procedure section: http://file.rdu.redhat.com/~ahoffer/2022/psa-clarifications/authentication/understanding-and-managing-pod-security-admission.html#security-context-constraints-psa-opting_understanding-and-managing-pod-security-admission

And updated the general info section to have just a basic note saying that system-created namespaces are permanently disabled, but that you can enable sync on other namespaces. (And then they'll just have to go to the other section for the specifics).   http://file.rdu.redhat.com/~ahoffer/2022/psa-clarifications/authentication/understanding-and-managing-pod-security-admission.html#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

Let me know what you think of this!

@stlaz If you can also weigh in on whether this approach is okay / whether my condensed note in the above section reads okay

xingxingxia commented:
/lgtm

stlaz commented:
lgtm

bergerhoffer commented:
/cherrypick enterprise-4.12

bergerhoffer commented:
/cherrypick enterprise-4.11
